### ISSUE TITLE
Inconsistency in MockFile.WriteAllText

### ISSUE THREAD

**Issue by manne** [tathamoddie/System.IO.Abstractions#52]
There is an inconsistency in MockFile.WriteAllText (and WriteAllLines), the method does not respect the encoding's preamble.
The expectation is, that there should be a BOM.

Example
```
static void Main()
{
	IFileSystem fs = new MockFileSystem();
	string mockPath = fs.Path.GetTempFileName();
	var mockContent = WriteAllText(fs, mockPath);
	
	IFileSystem fsr = new FileSystem();
	string tempPath = System.IO.Path.GetTempFileName();
	var realContent = WriteAllText(fsr, mockPath);
	
	if(!mockContent.Equals(realContent))
	{
		throw new Exception("Should have equal content");
	}
}

static byte[] WriteAllText(IFileSystem fs, string path)
{
	fs.File.WriteAllText(path, "using", Encoding.UTF8);
	var result = fs.File.ReadAllBytes(path);
	fs.File.Delete(path);
	return result;
}
```

**Comment by manne** [comment]
or #68

**Comment by manne** [comment]
fixed with #107
